scripts/dependency_graph.py

```
@@ -0,0 +1,109 @@
+            results[group_count].add(dependency)
+            new_matched_deps.add(dependency)
+            del dependencies[dependency]
+    matched_deps = matched_deps | new_matched_deps
+    if dependencies:
+        return make_groups(dependencies, group_count + 1, matched_deps, results)
+    return results
+
+
+def update_appset(groups: dict[int, set[str]], appset_path: pathlib.Path):
+    with appset_path.open('r') as fh:
+        appset = load(fh, Loader=Loader)
```

Ruff (0.14.4)

Probable use of unsafe loader `Loader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)